Src/FluentAssertions/Xml/XmlReaderValidator.cs

```diff
@@ -12,9 +13,32 @@ internal class XmlReaderValidator
         private readonly XmlReader subjectReader;
         private readonly XmlReader otherReader;
 
-        private string GetCurrentLocation() => "/" + string.Join("/", locationStack.Reverse());
+        private string GetCurrentLocation()
+        {
+            var keyBuilder = new StringBuilder();
+            var resultBuilder = new StringBuilder();
+            foreach (var location in locationStack.Reverse())
+            {
+                keyBuilder.Append($"/{location}");
+                resultBuilder.Append($"/{location}");
+                var locationCount = locationCounts[keyBuilder.ToString()];
+                if (locationCount > 1)
+                {
+                    resultBuilder.Append($"[{locationCount}]");
+                }
+            }
+
+            string result = resultBuilder.ToString();
+            if (string.IsNullOrEmpty(result))
```

Comment:
This can never be `null`, right?

Reply:
Right, it cannot be `null`, at most empty. 

Could also be written as
```c#
if (resultBuilder.Length == 0)
{
    return "/";
}

string result = resultBuilder.ToString();
return result;
```